test/jdk/java/awt/PopupMenu/PopupMenuLocation.java

```diff
@@ -109,19 +114,23 @@ private static void openPopup(final Frame frame) throws Exception {
         Robot robot = new Robot();
         robot.setAutoDelay(200);
         robot.waitForIdle();
-        Point pt = frame.getLocationOnScreen();
-        robot.mouseMove(pt.x + frame.getWidth() / 2, pt.y + 50);
+        final AtomicReference<Point> pt = new AtomicReference<>();
+        SwingUtilities.invokeAndWait(() -> {
+                                          pt.set(frame.getLocationOnScreen());
```

Comment:
This test uses only awt components which should work on any threads, why invokeAndWait is necessary?

Reply:
Removed from EDT